packages/sdk/src/evm/core/sdk.ts

```diff
@@ -159,10 +159,23 @@ export class ThirdwebSDK extends RPCConnectionHandler {
   public storage: ThirdwebStorage;
 
   constructor(
-    network: NetworkInput,
+    network: NetworkInput | Chain,
     options: SDKOptions = {},
     storage: ThirdwebStorage = new ThirdwebStorage(),
   ) {
+    if (
+      typeof network !== "string" &&
+      typeof network !== "number" &&
+      !ethers.Signer.isSigner(network) &&
+      !ethers.providers.Provider.isProvider(network)
```

Comment:
Wonder if it's safer to check if object and has rpc property in there with at least one rpc?

Reply:
I'm scared that `Signer` JS typeof would also be `object` (same for provider)